jgrapht-io/src/main/java/org/jgrapht/nio/tsplib/TSPLIBImporter.java

```diff
@@ -0,0 +1,735 @@
+     * @param p2 a two or three dimensional point
+     * @return the {@code ATT} edge weight for points p1 and p2
+     */
+    int compute2DPseudoEuclideanDistance(V p1, V p2)
+    {
+        double xd = elementGetter.getElement(p1, 0) - elementGetter.getElement(p2, 0);
+        double yd = elementGetter.getElement(p1, 1) - elementGetter.getElement(p2, 1);
+        double rij = Math.sqrt((xd * xd + yd * yd) / 10.0);
+        double tij = Math.round(rij);
+        if (tij < rij) {
+            return (int) (tij + 1);
+        } else {
+            return (int) tij;
+        }
+    }
+
+    private double getL1Distance(V p1, V p2)
+    {
+        double elementSum = 0;
+        for (int i = 0; i < vectorLength; i++) {
+            double delta = elementGetter.getElement(p1, i) - elementGetter.getElement(p2, i);
+            elementSum += Math.abs(delta);
+        }
+        return elementSum;
+    }
+
+    private double getL2Distance(V p1, V p2)
+    {
+        double elementSum = 0;
+        for (int i = 0; i < vectorLength; i++) {
+            double delta = elementGetter.getElement(p1, i) - elementGetter.getElement(p2, i);
+            elementSum += delta * delta;
+        }
+        return Math.sqrt(elementSum);
+    }
+
+    private double getLInfDistance(V p1, V p2)
+    {
+        double maxElement = 0;
+        for (int i = 0; i < vectorLength; i++) {
+            double delta = elementGetter.getElement(p1, i) - elementGetter.getElement(p2, i);
+            maxElement = Math.max(maxElement, Math.abs(delta));
+        }
+        return maxElement;
+    }
+
+    // node build
+
+    private V createVector2D(String line)
+    {
+        String[] elements = splitLineElements(line, 3);
+        // index of the point in elements[0] is not necessary
+        double x = Double.parseDouble(elements[1]);
+        double y = Double.parseDouble(elements[2]);
+        return vectorFactory.apply(new double[] { x, y });
+    }
+
+    private V createVector3D(String line)
+    {
+        String[] elements = splitLineElements(line, 4);
+        // index of the point in elements[0] is not necessary
+        double x = Double.parseDouble(elements[1]);
+        double y = Double.parseDouble(elements[2]);
+        double z = Double.parseDouble(elements[3]);
+        return vectorFactory.apply(new double[] { x, y, z });
+    }
+
+    private static String[] splitLineElements(String line, int expectedLineElements)
+    {
+        String[] elements = line.split(" ");
+        if (elements.length != expectedLineElements) {
+            throw new IllegalArgumentException(
+                "Unexpected number of line elements: " + elements.length + " in line: " + line);
+        }
+        return elements;
+    }
+
+    // read of coordinates and graph computation
+
+    /**
+     * Read the complete {@link Graph} of node locations and returns true if all locations are
+     * distinct.
+     */
+    private boolean readCompleteNodeGraphData(
```

Comment:
Please merge methods `readCompleteNodeGraphData`, `readNodeCoordinateData` and `buildCompleteGraph` into a single method called 'readGraph` or something like this.

I know that best practices are to keep methods small, but in this case it is over complicating things.